tfx/dsl/component/experimental/annotations.py

```diff
@@ -188,10 +196,5 @@ class BeamComponentParameter(_PipelineTypeGeneric):
   pass
 
 
-# TODO(ccy): potentially make this compatible `typing.TypedDict` in
-# Python 3.8, to allow for component return value type checking.
-class OutputDict:
-  """Decorator declaring component executor function outputs."""
-
-  def __init__(self, **kwargs):
-    self.kwargs = kwargs
+class OutputDict(TypedDict):
```

Comment:
I doubt this work? TypedDict annotations should be specified statically in order to work properly. The right direction is not to upgrade the existing OutputDict, but to support both OutputDict and TypedDict.